### PR TITLE
Jhh/linkify last line correctly

### PR DESCRIPTION
**Pull request by jasonhawkharris**
## Test plan
1. Check manually that last line of group of links is both clickable and going pointing to the correct URL.

### CODE REVIEW

File: client/web/src/repo/blob/codemirror/links.ts [hunk 1]
```diff
@@ -103,7 +103,7 @@ class LinkBuilder implements PluginValue {
                         lineDecorations = []
 
                         const start = line.from + occurrence.range.start.character
-                        const end = line.from + occurrence.range.end.character
+                        const end = start + line.length
```
Comment (fkling) [inline client/web/src/repo/blob/codemirror/links.ts:106]: Do you have an example to reproduce the issue? At first glance, extracting links from the whole line doesn't seem correct, though I guess it won't be a problem in practice.

Comment (jasonhawkharris) [reply client/web/src/repo/blob/codemirror/links.ts:106]: The issue [here](https://github.com/sourcegraph/sourcegraph/issues/57043) has a good example of where and when this is happening. I was able to reproduce locally as well.

**Comment by fkling** [comment]
After looking at the LSIF/SCIP data I actually think it's a problem with the data itself, not the logic. If you look at other YAML files, each of them has their last character highlighted incorrectly (or rather, not at all). So there seems to be a general off-by-one error for the last token in a YAML file when generating the data.

(also when the line contains an emoji btw)

**Comment by jasonhawkharris** [comment]
@fkling I agree with you that this is an LSIF/SCIP issue, but for the sake of the customer who seems to be concerned about this, I think this would work as a temporary workaround. Then we can inform the graph team of the issue and hopefully get the bigger issue solved.

File: client/web/src/repo/blob/codemirror/links.ts [hunk 2]
```diff
@@ -103,7 +103,10 @@ class LinkBuilder implements PluginValue {
                         lineDecorations = []
 
                         const start = line.from + occurrence.range.start.character
-                        const end = line.from + occurrence.range.end.character
+                        // There is likely something wrong with the LSIF/SCIP data that is causing
+                        // The last character in a file to go unhighlighted. This is a temporary workaround
+                        // and shouldn't be necessary when this is fixed on the code intel side.
+                        const end = start + line.length
```
Comment (peterguy) [inline client/web/src/repo/blob/codemirror/links.ts:109]: does this address the one-character-short decorations on non-links also?

**Comment by jasonhawkharris** [comment]
Closing because this is a better fix: https://github.com/sourcegraph/sourcegraph/pull/57343

cc: @fkling

**Pull request closed without merging**